src/averageAges.js

```diff
@@ -36,8 +45,22 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = !!withChildren
+    ? people.filter(person =>
+      person.sex === 'f' && people.some(child => child.mother === person.name))
+    : people.filter(person => person.sex === FEMALE);
```

Comment:
The logic here is incorrect. If `withChildren` is specified, you need to check for it inside the ternary operator and then for the `FEMALE` value. Fix the logic.